react_demo/cpp/index.html

```diff
@@ -0,0 +1,250 @@
+              '>=': [ 'More than or equal to {rhs} {var}', '<' ],
+              '<=': [ 'Less than or equal to {rhs} {var}', '>' ]
+            };
+
+            // Sort so that longer strings match first.
+            var opnames = Object.keys(ops).sort().reverse();
+
+            // re = /(variable name)(one of operators)(variable value)/
+            var re = new RegExp('(.+)((?:' + opnames.join(')|(?:') + '))(.+)');
+
+            // Test for comparison expression. Return if humanized.
+            var match = re.exec(expr);
+            var op;
+            if (match && (op = ops[match[2]])) {
+              return (yes ? op[0] : ops[op[1]][0] ).replace('{var}', match[1]).replace('{rhs}', match[3]);
+            }
+
+            // Test for boolean expression. Return if humanized.
+            if (expr.indexOf('Has') === 0) {
+              return (yes
+                ? expr.replace(/^Has/, 'Has ')
+                : expr.replace(/^Has/, 'Does not have ')
+              );
+            }
+
+            // Cannot humanize.
+            return expr;
+          }
+        });
+      });
+      ;(function (root, factory) {
+        root.InsightsBrowser = factory(
+          root.React,
+          root.InsightsNavigation,
+          root.MutualInformationInsightsVisualizer
+        );
+      })(this, function (React, InsightsNavigation, MutualInformationInsightsVisualizer) {
+        return React.createClass({
+          render: function () {
+            var _this = this;
+
+            // HACK(sompylasar): Simplest error handling.
+            try {
+              var data = _this.props.data;
+
+              var navData = data.data;
+
+              var insightCereal = data.data.insight;
+              var insightType = insightCereal.polymorphic_name;
+              var insightData = insightCereal.ptr_wrapper.data;
+
+              var insightTypeToVisualizerMap = {
+                "insight::MutualInformation": MutualInformationInsightsVisualizer
+              };
+
+              var Visualizer = insightTypeToVisualizerMap[insightType];
+
+              if ( !Visualizer ) {
+                return <div className="c5t-insights-browser">
+                  <InsightsNavigation navData={navData} />
+                  <div className="c5t-error">
+                    No visualizer found for `<span>{insightType}</span>`.
+                  </div>
+                </div>;
+              }
+
+              return <div className="c5t-insights-browser">
+                <InsightsNavigation navData={navData} />
+                <Visualizer insightData={insightData} />
+              </div>;
+            }
+            catch (ex) {
+              return <div className="c5t-insights-browser">
+                <div className="c5t-error">
+                 An error occurred: <span>{ex.message}</span>
+                </div>
+              </div>;
+            }
+          }
+        });
+      });
+      ;(function () {
+        var DATA_JSON;
```

Comment:
Rename similarly to `PAGE_TITLE`: `DATA_JSON_FROM_C_PLUS_PLUS`.

BTW, this line could also be removed if there will be no references to `DATA_JSON_FROM_C_PLUS_PLUS` after putting in the JSON.